src/tau_clients/nsx_defender/__init__.py

```diff
@@ -626,17 +668,128 @@ def get_completed(
         )
         return self.get("analysis", "completed", params=params)
 
-    def get_result(self, uuid: str) -> Dict[str, Any]:
+    def get_result(
+        self,
+        uuid: str,
+        report_uuid: Optional[str] = None,
+        include_report: Optional[bool] = True,
+    ) -> Dict[str, Any]:
         """
-        Get report results for a given task.
+        Get the result for a given task.
 
         :param str uuid: the unique identifier of the submitted task
+        :param str|None report_uuid: if specified, include this specific report
+        :param bool include_report: whether to include a report
         :rtype: dict[str, any]
-        :return: a dictionary containing the analysis report
+        :return: the result of an analysis
         """
-        params = {"uuid": uuid}
+        params = tau_clients.purge_none(
+            {
+                "uuid": uuid,
+                "report_uuid": report_uuid,
+                "full_report_score": 0 if include_report else -1,
+            }
+        )
         return self.get("analysis", "get", params=params)
 
+    def get_result_process_snapshot_names(self, uuid: str) -> List[Dict[str, str]]:
+        """
+        Get all the process snapshot names for a given task.
+
+        :param str uuid: the unique identifier of the submitted task
+        :rtype: list[dict[str, str]]
+        :return: a list of dictionaries like:
+            [
+                {
+                    "task_uuid": "aba8dccf1641001000c9dd8526ac7c9d",
+                    "report_uuid": "4ffa644851ebe118wu1pbwlt3C44cwAYRtfdXY5L3gRhHJ_QANAyuQ",
+                    "artifact_name": "process_snapshots_2",
+                },
+                ...
+            ]
+        """
+        return self._get_result_artifact_names(
+            uuid=uuid, report_types=["ll-int-win"], metadata_types=["process_snapshot"]
+        )
+
+    def get_result_artifact_names(
+        self,
+        uuid: str,
+        report_types: Optional[List[str]] = None,
+        metadata_types: Optional[List[str]] = None,
+    ) -> List[Dict[str, str]]:
+        """
+        Get all the artifact names for a given task.
+
+        :param str uuid: the unique identifier of the submitted task
+        :param list[str]|None report_types: optional list of report types to filter on
+        :param list[str]|None metadata_types: optional list of metadata types to filter on
+        :rtype: list[dict[str, str]]
+        :return: a list of dictionaries like:
+            [
+                {
+                    "task_uuid": "aba8dccf1641001000c9dd8526ac7c9d",
+                    "report_uuid": "4ffa644851ebe118wu1pbwlt3C44cwAYRtfdXY5L3gRhHJ_QANAyuQ",
+                    "artifact_name": "process_snapshots_2",
+                },
+                ...
+            ]
+        """
+        ret = []
+        if not report_types:
+            report_types = set([])
+        if not metadata_types:
+            metadata_types = set([])
```

Comment:
In the parameter type annotation, these two variables are supposed to be lists but here they are created as sets. Which one is the correct type?

Reply:
Replaced with iterable